plugins/database/cassandra/cassandra.go

```diff
@@ -80,162 +71,174 @@ func (c *Cassandra) getConnection(ctx context.Context) (*gocql.Session, error) {
-	c.Lock()
-	defer c.Unlock()
+func (c *Cassandra) UpdateUser(ctx context.Context, req newdbplugin.UpdateUserRequest) (newdbplugin.UpdateUserResponse, error) {
+	if req.Password == nil && req.Expiration == nil {
+		return newdbplugin.UpdateUserResponse{}, fmt.Errorf("no changes requested")
+	}
 
-	statements = dbutil.StatementCompatibilityHelper(statements)
+	if req.Password != nil {
+		err := c.changeUserPassword(ctx, req.Username, req.Password)
+		return newdbplugin.UpdateUserResponse{}, err
+	}
+	// Expiration is no-op
+	return newdbplugin.UpdateUserResponse{}, nil
+}
 
+func (c *Cassandra) changeUserPassword(ctx context.Context, username string, changePass *newdbplugin.ChangePassword) error {
 	session, err := c.getConnection(ctx)
 	if err != nil {
 		return err
 	}
 
-	revocationCQL := statements.Revocation
-	if len(revocationCQL) == 0 {
-		revocationCQL = []string{defaultUserDeletionCQL}
+	rotateCQL := changePass.Statements.Commands
+	if len(rotateCQL) == 0 {
+		rotateCQL = []string{defaultChangePasswordCQL}
 	}
 
 	var result *multierror.Error
-	for _, stmt := range revocationCQL {
+	for _, stmt := range rotateCQL {
 		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
 			query = strings.TrimSpace(query)
 			if len(query) == 0 {
 				continue
 			}
 
-			err := session.Query(dbutil.QueryHelper(query, map[string]string{
+			m := map[string]string{
 				"username": username,
-			})).WithContext(ctx).Exec()
-
+				"password": changePass.NewPassword,
+			}
+			err := session.
+				Query(dbutil.QueryHelper(query, m)).
+				WithContext(ctx).
+				Exec()
 			result = multierror.Append(result, err)
 		}
 	}
 
 	return result.ErrorOrNil()
 }
 
-func (c *Cassandra) RotateRootCredentials(ctx context.Context, statements []string) (map[string]interface{}, error) {
-	// Grab the lock
+// DeleteUser attempts to drop the specified user.
+func (c *Cassandra) DeleteUser(ctx context.Context, req newdbplugin.DeleteUserRequest) (newdbplugin.DeleteUserResponse, error) {
 	c.Lock()
 	defer c.Unlock()
 
 	session, err := c.getConnection(ctx)
 	if err != nil {
-		return nil, err
-	}
-
-	rotateCQL := statements
-	if len(rotateCQL) == 0 {
-		rotateCQL = []string{defaultRootCredentialRotationCQL}
+		return newdbplugin.DeleteUserResponse{}, err
 	}
 
-	password, err := c.GeneratePassword()
-	if err != nil {
-		return nil, err
+	revocationCQL := req.Statements.Commands
+	if len(revocationCQL) == 0 {
+		revocationCQL = []string{defaultUserDeletionCQL}
 	}
 
 	var result *multierror.Error
-	for _, stmt := range rotateCQL {
+	for _, stmt := range revocationCQL {
 		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
 			query = strings.TrimSpace(query)
 			if len(query) == 0 {
 				continue
 			}
 
-			err := session.Query(dbutil.QueryHelper(query, map[string]string{
-				"username": c.Username,
-				"password": password,
-			})).WithContext(ctx).Exec()
+			m := map[string]string{
+				"username": req.Username,
+			}
+			err := session.
```

Comment:
This is purely stylistic, but I don't think we tend to break chained function calls into multiple lines unless that's broken out by a multi-line struct declaration in the function's input param. Breaking out object's instantiation field values into multiple lines, like with `m` above, is fine though.

Reply:
I ended up doing this because the line was getting fairly long with a lot of parenthesis. I can change it back if you want though.